src/main/kotlin/org/malachite/estella/services/InterviewService.kt

```diff
@@ -1,23 +1,130 @@
+        val application = applicationRepository.findById(applicationId)
+        application.ifPresent {
+            val interview = Interview(null, null, minutesLength, application.get(), listOf(), setOf())
+            interviewRepository.save(interview)
+            mailService.sendInterviewInvitationMail(offerService.getOfferByApplication(it), interview)
+        }
+    }
+
+
+    fun getInterview(id: UUID): Interview = withExceptionThrower { interviewRepository.findById(id).get() }
+
+    fun getUserFromInterviewUuid(interviewId: UUID): User? =
+        getInterview(interviewId).application.jobSeeker.user
+
+    fun getOrganizationFromPartner(hrPartnerId: Int): Organization =
+            hrPartnerService.getHrPartner(hrPartnerId).organization
+
+    fun getLastInterviewFromApplicationId(applicationId: Int): InterviewId =
+            withExceptionThrower { interviewRepository.getAllByApplicationId(applicationId).sortedWith { a, b ->
+                a.dateTime?.compareTo(b.dateTime) ?: -1
+            }.first() }.toId()
+
+    fun setHosts(id: UUID, hostsMails: List<String>) {
+        if (!canHrUpdate(id)) throw UnauthenticatedException()
+        val interview = interviewRepository.findById(id)
+        interview.ifPresent {
+            val int = interviewRepository.save(it.copy(hosts=hostsMails))
+            val application = int.application
+            val offer = recruitmentProcessService.getProcessFromStage(application.stage).offer
+            hostsMails.forEach { mail -> mailService.sendInterviewDevInvitationMail(offer, int, application, mail) }
+        }
+    }
+
+    fun setDate(id: UUID, dateTime: Timestamp) {
+        if (!canJobSeekerUpdate(id)) throw UnauthenticatedException()
+        val interview = interviewRepository.findById(id)
+        interview.ifPresent {
+            val int = interviewRepository.save(it.copy(dateTime = dateTime))
+            val application = it.application
+            val offer = recruitmentProcessService.getProcessFromStage(application.stage).offer
+            mailService.sendInterviewDateConfirmationMail(offer, int, application, application.jobSeeker.user.mail)
+            int.hosts.forEach { mail -> mailService.sendInterviewDateConfirmationMail(offer, int, application, mail) }
+        }
+    }
+
+    fun setNotes(id: UUID, notes: Set<InterviewNote> ) {
+        if (!canHrUpdate(id)) throw UnauthenticatedException() //TODO dev security?
+        val interview = interviewRepository.findById(id)
+        interview.ifPresent {
+            val persNotes = mutableSetOf<InterviewNote>()
+            notes.forEach { persNotes.add(interviewNoteRepository.save(it)) }
+            interviewRepository.save(it.copy(notes = persNotes))
+        }
+    }
+
+    fun canHrUpdate(id: UUID?): Boolean {
+        val userDetails = securityService.getUserDetailsFromContext()
+        if (securityService.isCorrectApiKey(userDetails?.token))
```

Comment:
Tego ifa możesz dać wcześniej. Wiem, że nie przejmujemy się performance'm, ale nie wiem, jak ciężkie jego to get user details - jeśli możemy to ominąć, to why not?